File: dataset/human/c/file_9742.c
#include<stdio.h>

#define N 100000
#define NIL -1

int depth(int);
void degree(int);

struct Node{
    int parent;
    int left;
    int right;
};

struct Node T[N];
int count=0;

int main(){

    int n,i,j,id,k,child,tmp;

    scanf("%d",&n);

    for(i=0;i<n;i++) T[i].parent=T[i].left=T[i].right=NIL;
    
    for(i=0;i<n;i++){

        scanf("%d%d",&id,&k);

        for(j=0;j<k;j++){

            scanf("%d",&child);

            T[child].parent=id;

            if(j==0) T[id].left=child;

            else T[tmp].right=child;

            tmp=child;

        }

    }

    for(i=0;i<n;i++){

        count=0;

        printf("node %d: parent = %d, depth = %d, ",i,T[i].parent,depth(i));

        if(T[i].parent==NIL) printf("root, [");

        else if(T[i].left==NIL) printf("leaf, [");

        else printf("internal node, [");

        count=0;

        degree(T[i].left);

        printf("]\n");

    }

    return 0;

}

int depth(int x){

    if(T[x].parent==NIL) return count;

    else{

    count++;

    return depth(T[x].parent);

    }

}

void degree(int x){

    if(x != NIL){

        if(count != 0) printf(", ");

        printf("%d",x);

        count++;

        degree(T[x].right);

    }

}

File: dataset/human/java/file_9742.java
import java.util.*;

public class Main {
	public static void main (String[] args) {
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		Node[] nodes = new Node[n];
		for (int i = 0; i < n; i++) {
		    nodes[i] = new Node(i);
		}
		for (int i = 0; i < n; i++) {
		    int x = sc.nextInt();
		    int left = sc.nextInt();
		    int right = sc.nextInt();
		    nodes[x].left = left;
		    nodes[x].right = right;
		    if (left >= 0) {
		        nodes[left].sibling = right;
		        nodes[left].parent = x;
		    }
		    if (right >= 0) {
		        nodes[right].sibling = left;
		        nodes[right].parent = x;
		    }
		}
		for (Node x : nodes) {
		    if (x.parent == -1) {
		        set(nodes, x.idx, 0);
		        break;
		    }
		}
		StringBuilder sb = new StringBuilder();
		for (Node x : nodes) {
		    sb.append(x).append("\n");
		}
		System.out.print(sb);
	}
	
	static int set(Node[] nodes, int idx, int depth) {
	    nodes[idx].depth = depth;
	    int max = 0;
	    if (nodes[idx].left >= 0) {
	        max = Math.max(max, set(nodes, nodes[idx].left, depth + 1));
	    }
	    if (nodes[idx].right >= 0) {
	        max = Math.max(max, set(nodes, nodes[idx].right, depth + 1));
	    }
	    nodes[idx].height = max;
	    return max + 1;
	}
	
	static class Node {
	    int idx;
	    int parent;
	    int depth;
	    int height;
	    int sibling;
	    int left;
	    int right;

	    public Node(int idx) {
	        this.idx = idx;
	        parent = -1;
	        sibling = -1;
	        left = -1;
	        right = -1;
	    }
	    
	    private int getDegree() {
	        int ans = 0;
	        if (left >= 0) {
	            ans++;
	        }
	        if (right >= 0) {
	            ans++;
	        }
	        return ans;
	    }
	    
	    public String toString() {
	        StringBuilder sb = new StringBuilder();
	        sb.append("node ").append(idx).append(": parent = ");
	        sb.append(parent).append(", sibling = ").append(sibling).append(", degree = ").append(getDegree());
	        sb.append(", depth = ").append(depth).append(", height = ").append(height).append(", ");
	        if (parent == -1) {
	            sb.append("root");
	        } else if (getDegree() == 0) {
	            sb.append("leaf");
	        } else {
	            sb.append("internal node");
	        }
	        return sb.toString();
	    }
	}
}
